base/array.jl

```diff
@@ -1365,6 +1365,7 @@ missing
 
 julia> popat!(a, 4)
 ERROR: BoundsError: attempt to access 3-element Vector{Int64} at index [4]
+Valid indices are 1:3.
```

Comment:
I agree with the suggestion to only do something extra when indices are non-standard. But if something more is added, perhaps the default BoundsError message should instead be 
```suggestion
ERROR: BoundsError: attempt to access 3-element Vector{Int64} with axes $(axes(...)) at index [4]
```

Reply:
+1 for adding things into the running text, just like the existing index hints for strings:

```julia
julia> "fαb"[3]
ERROR: StringIndexError: invalid index [3], valid nearby indices [2]=>'α', [4]=>'b'
```